x/ibc/04-channel/types/packet.go

```diff
@@ -1,49 +1,45 @@
 package types
 
 import (
-	"encoding/binary"
-
 	"github.com/tendermint/tendermint/crypto/tmhash"
 
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
-// CommitPacket appends bigendian encoded timeout height and commitment bytes
-// and then returns the hash of the result bytes.
+// CommitPacket return the hash of commitment bytes
 // TODO: no specification for packet commitment currently,
 // make it spec compatible once we have it
-func CommitPacket(data exported.PacketDataI) []byte {
-	buf := make([]byte, 8)
-	binary.BigEndian.PutUint64(buf, data.GetTimeoutHeight())
-	buf = append(buf, data.GetBytes()...)
-	return tmhash.Sum(buf)
+func CommitPacket(data []byte) []byte {
```

Comment:
We also need to commit packet timeout on sender chain.

Either this should be hashed in with the data. Or it should be committed seperately

Previously, the timeout height was part of packetdataI. Now that they are seperated, we still need a way to commit the timeout, and be able to prove it was a committed value on the receiving chain

Reply:
Okay, that makes sense.  I'll make it work as before but pass in the timeout height as an argument.

Thanks for noticing this!

Reply:
Ah yes, I missed this, thanks @AdityaSripal, precisely correct.